crates/orderbook/src/orderbook.rs

```diff
@@ -340,11 +374,7 @@ impl Orderbook {
         // Verify that the new order is a valid replacement order by checking
         // that the `app_data` encodes an order cancellation and that both the
         // old and new orders have the same signer.
-        let cancellation = OrderCancellation {
-            order_uid: old_order.metadata.uid,
-            ..Default::default()
-        };
-        if new_order.data.app_data != cancellation.hash_struct()
+        if app_data.metadata.original_order != old_order.metadata.uid
```

Comment:
According to grafana this endpoint was never called successfully in the last 90 days. Since nobody is using it I think we might as well break the API completely and remove the `order_uid` from the path.
Since we now require the uid to be present in the appdata there is no need to also have it in the path and that also removes 1 extra error case.